### DISCUSSION TITLE
Encapsulate as an API?

### DISCUSSION THREAD

**Discussion by silvacarl2** [ggml-org/whisper.cpp#364] [Bindings]
any recommendations on encapsulating this as an API using something like FastAPI?

**Comment by silvacarl2** [comment]
this seems like something that could be easily added: https://github.com/haseeb-heaven/FastServer?ref=morioh.com&utm_source=morioh.com

if we have the time we will add it.

**Comment by ggerganov** [comment]
Not sure what you mean by "encapsulating this as an API" - do you need an HTTP RESTful API for `whisper.cpp`?
If so, there are many ways to create one - depends what you want to do with it

**Comment by silvacarl2** [comment]
yes.  we know python, nodejs, php eally well.  not quite as sure about cpp yet.  but the idea would be to be able to call your whisper.cpp as an API from python, or nodejs, or php as a RESTFUL API or as a direct call from one of those languages.  like this: https://morioh.com/p/189da50601a3

**Reply by ilyazub** [reply]
Here's how `whisper.cpp` can be used with Python: https://huggingface.co/spaces/RASMUS/Whisper-youtube-crosslingual-subtitles/blob/main/app.py#L255-L261. It simply uses `os.system("./whisper.cpp/main ...")`.

Author: @R4ZZ3 (https://github.com/openai/whisper/discussions/283)

**Comment by silvacarl2** [comment]
super cool, will try this out!

**Comment by jackleeforce** [comment]
> Here's how `whisper.cpp` can be used with Python: https://huggingface.co/spaces/RASMUS/Whisper-youtube-crosslingual-subtitles/blob/main/app.py#L255-L261. It simply uses `os.system("./whisper.cpp/main ...")`.
> 
> Author: @R4ZZ3 ([openai/whisper#283](https://github.com/openai/whisper/discussions/283))

this is good way to using Whisper.cpp in Python, it's easy to wrap as a service, but how to avoid load model each time?
